modules/quant_search/src/Form/SearchEntitiesForm.php

```diff
@@ -242,9 +244,100 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       ];
     }
 
+    // Taxonomy configuration.
+    $form['quant_search_entity_taxonomy_markup'] = [
+      '#type' => 'markup',
+      '#markup' => '<h2>Taxonomy configuration</h2>',
+      '#suffix' => '<hr/>',
+    ];
+
+    // Taxonomy configuration.
+    $form['quant_search_entity_taxonomy_term'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Taxonomy terms'),
+      '#description' => $this->t('Keep search records for taxonomy terms updated.'),
+      '#default_value' => $config->get('quant_search_entity_taxonomy_term'),
+    ];
+
     return parent::buildForm($form, $form_state);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+
+    // Get all node type ids. Add 'default' to handle default values.
+    $nodeTypeIds = ['default'];
+    $nodeTypes = \Drupal::entityTypeManager()
+      ->getStorage('node_type')
+      ->loadMultiple();
+    foreach ($nodeTypes as $nodeType) {
+      $nodeTypeIds[] = $nodeType->id();
+    }
+
+    // Get all tokens.
+    $tokens = [];
+    foreach ($nodeTypeIds as $nodeTypeId) {
+      $typeTokens = $form_state->getValue('search_tokens_node_' . $nodeTypeId);
+      // Only check tokens if default or node type is overriding the defaults.
+      if ($nodeTypeId === 'default' || $typeTokens['enabled']) {
+        $tokens[] = trim($typeTokens['quant_search_title_token']);
+        $tokens[] = trim($typeTokens['quant_search_summary_token']);
+        $tokens[] = trim($typeTokens['quant_search_image_token']);
+      }
+    }
+
+    // Token module will find some of the invalid tokens.
+    $invalidTokens = \Drupal::token()->getInvalidTokensByContext(implode(' ', $tokens), ['all']);
```

Comment:
We may need a `quant_search` update hook to enable the token module for a clean upgrade pathway.

If someone were to update an existing install without token installed they would get `Call to undefined method Drupal\Core\Utility\Token::getInvalidTokensByContext()` errors

Reply:
Good catch, I'll add.